Install asym-crypto-yaml into go-agent (OPS-4261)

nadeemshahzad opened the pull request:
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

coryleeio commented on docker/build/go-agent/Dockerfile:
```diff
@@ -85,6 +85,13 @@ RUN apt-get update && apt-get install zlib1g-dev wget lsb-core -qy && cd /opt &&
+python3.6 get-pip.py
+
+# Install asym-crypto-yaml to decrypt encrypted yaml config files
+pip3.6 install asym-crypto-yaml
```
I think you just want to add this to an appropriate .in file the configuration/requirements and run make upgrade

nadeemshahzad replied:
@coryleeio I try to run make upgrade after adding asym-crypto-yaml to new ini file, but it failed as we are using pip-compile with python2.7 for all packages and asym-crypto-yaml requires python >=3.6. 
```pip-compile --upgrade -o requirements3.6.txt requirements/asym-crypto-yaml.in
Could not find a version that matches asym-crypto-yaml (from -r requirements/asym-crypto-yaml.in (line 2))
No versions found
Was https://pypi.python.org/simple reachable?
```
Can I add this requirement to tubular in new requirements3.6.txt file and add new task in pipeline to install this using pip3.6 or pre-build go-agent having this requirement installed ?

coryleeio replied:
Yeah I suppose the extra requirements3.txt file would be better than this as otherwise the configuration playbook calls out to something that may or may not actually be there.  Atleast this way the play would install it even if it is less than ideal.

Another option would be adding python 2.7 to tox and verifying outside of docker in a normal venv that the tests pass  in python2 for asym-crypto-yaml.  If they do we *might* be able to relax the version constraint.  I would do one of those things.. Given that python 2 is almost not supported lets go with the requirements3.txt idea I suppose. Thanks!

nadeemshahzad commented:
@coryleeio I have updated tox and setup.py to run tests with python 2.7 and python 3.5 , tests are failing.  I have updated configuration PR ( https://github.com/edx/configuration/pull/5346 ) to add new requirement file ```asym-crypto-yaml.txt``` that we can use to install asym-crypto-yaml in pipelines using pip3.6.
